src/server/pfs/server/driver.go

```diff
@@ -476,17 +513,43 @@ func (d *driver) FinishCommit(ctx context.Context, commit *pfs.Commit) error {
 			if err := tree.DeleteFile(filePath); err != nil {
 				// Deleting a non-existent file in an open commit should
 				// be a no-op
-				if !(hashtree.Code(err) == hashtree.PathNotFound) {
+				if hashtree.Code(err) != hashtree.PathNotFound {
 					return err
 				}
 			}
 		} else {
-			var objectHash string
-			var size int64
-			fmt.Sscanf(string(kv.Value), "%d %s", &size, &objectHash)
-			if err := tree.PutFile(filePath, []*pfs.Object{{Hash: objectHash}}, size); err != nil {
+			records := &PutFileRecords{}
+			if err := proto.Unmarshal(kv.Value, records); err != nil {
 				return err
 			}
+			if !records.Split {
```

Comment:
In that case you'd get 2 records, there's a 1-to-1 correspondence between records and `put-file` requests.